daemon/audit.c

```diff
@@ -312,6 +328,31 @@ audit_record_from_textfile_new(const char *filename, bool purge)
 	return (AuditRecord *)record;
 }
 
+static bool
+audit_log_check_delimiter(const char *log_file)
+{
+	int fd = -1;
+	char buf[sizeof(AUDIT_DELIMITER)];
+	size_t log_file_size = 0;
+	bool ret = false;
+
+	IF_TRUE_RETVAL(0 >= (log_file_size = file_size(log_file)), true);
+	IF_TRUE_RETVAL(0 > (fd = open(log_file, O_RDONLY)), true);
+
+	IF_TRUE_GOTO(-1 == lseek(fd, -strlen(AUDIT_DELIMITER), SEEK_END), out);
+	IF_TRUE_GOTO(-1 == fd_read(fd, buf, strlen(AUDIT_DELIMITER)), out);
+
+	buf[sizeof(AUDIT_DELIMITER) - 1] = '\0';
```

Comment:
This should be buf[sizeof(AUDIT_DELIMITER)] to maintain the newline for comparision